apps/fz_wall/lib/fz_wall/cli/live.ex

```diff
@@ -71,23 +71,28 @@ defmodule FzWall.CLI.Live do
   Deletes nftables rule.
   """
   def delete_rule(params) do
-    rule_str = rule_str(params)
+    rule_str(params)
+    |> delete_rule_matching()
+  end
+
+  def delete_rules(source) do
+    source_match_str(source)
+    |> delete_rule_matching()
+  end
+
+  defp delete_rule_matching(rule_str) do
     rules = exec!("#{nft()} -a list table inet #{@table_name}")
 
-    case rule_handle_regex(~r/#{rule_str}.*# handle (?<num>\d+)/, rules) do
+    # When a rule is deleted the others might change handle so we need to
+    # re-scan each time.
+    case rule_handle_regex(~r/^\s*#{rule_str}.*# handle (?<num>\d+)/m, rules) do
       nil ->
-        raise("""
-          ######################################################
-          Could not get handle to delete rule!
-          Rule spec: #{rule_str}
-
-          Current rules:
-          #{rules}
-          ######################################################
-        """)
+        :no_rule
```

Comment:
Would it be worth doing ` Logger.warn("Could not get handle to delete rule: #{rule_str}")` here?

May help a user or two hitting issues in case this code path is hit?

Reply:
Oh wait, I guess this is changing with the named sets implementation, so nvmd

Reply:
The problem here is that this would be logged every time a device is deleted and I think that might confuse users.

I could keep track of the number of rules found, so if it's 0 I log the message or raise the exception but I think it will make the code messier and since this **will** be gone with firezone/product#398 I think it's not worth the change.